database/migrations/2024_06_19_074908_modify_users_table.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('users', function (Blueprint $table) {
+            # add : role_id column
+            $table->unsignedBigInteger('role_id')
+                ->default(2)
+                ->comment('admin:1 user:2')
+                ->after('password');
+            # add : avatar column
+            $table->longText('avatar')
```

Comment:
please use varchar for avatar, also if you plan to upload the avatar image use this approach where we don't actually upload the base64 file format of the file but just the filename, we will upload the file from our asset folder

```
namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Illuminate\Support\Str;

class FileUploadController extends Controller
{
    public function upload(Request $request)
    {
        // Validate the request
        $request->validate([
            'file' => 'required|mimes:jpg,jpeg,png,bmp|max:2048',
        ]);

        // Handle the file upload
        if ($request->file('file')) {
            $file = $request->file('file');

            // Generate a custom filename
            $filename = Str::random(10) . '.' . $file->getClientOriginalExtension();

            // Store the file with the custom filename
            $path = $file->storeAs('uploads', $filename, 'public');

            return back()->with('success', 'File uploaded successfully.')->with('file', $path);
        }

        return back()->with('error', 'Please select a file to upload.');
    }
}



```

Reply:
I understand what you say.
Teacher Ditzler has said I should use base64.
I don't have any idea which I should choose.  I want to discuss with you about it.

Reply:
oh, okay .. please proceed with base64, will merge this